src/core/util/util.js

```diff
@@ -799,6 +799,23 @@ function UtilFactory($document, $timeout, $compile, $rootScope, $$mdAnimate, $in
 
         return start + change * (-2 * tc + 3 * ts);
       }
+    },
+
+    /**
+     * When used with the Array.filter() method; provides an easy mechanism for removing duplicates
+     * from an array.
+     *
+     *    var myArray = [1, 2, 2, 3, 3, 3, 4, 4, 4, 4];
+     *    myArray.filter($mdUtil.uniq) => [1, 2, 3, 4]
+     *
+     * @param value
+     * @param index
+     * @param self
+     *
+     * @returns {boolean}
+     */
+    uniq: function(value, index, self) {
+      return self.indexOf(value) === index;
```

Comment:
@crisbeto You mean something like `$mdUtil.uniq(myArray)` and have it call the `filter()` function internally? I guess so. I'll update the code.